lib/auth/rotate.go

```diff
@@ -400,6 +406,97 @@ func (a *Server) autoRotate(ca types.CertAuthority) error {
 	return nil
 }
 
+type CertAuthorityMap = map[types.CertAuthType]types.CertAuthority
+
+// getAllCertificates returns all certificates authorities including private keys.
+func (a *Server) getAllCertificates(ctx context.Context, clusterName string) (CertAuthorityMap, error) {
+	certs := make(CertAuthorityMap)
+
+	for _, caType := range types.CertAuthTypes {
+		ca, err := a.Trust.GetCertAuthority(ctx, types.CertAuthID{
+			Type:       caType,
+			DomainName: clusterName,
+		}, true)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		certs[caType] = ca
+	}
+
+	return certs, nil
+}
+
+// findDuplicatedCertificates checks if any CA provided as caTypes has a duplicate in allCerts. List of all duplicates
+// is returned as a result. If no duplicates are found, then caTypes is returned in unmodified form.
+func findDuplicatedCertificates(caTypes []types.CertAuthType, allCerts CertAuthorityMap) []types.CertAuthType {
```

Comment:
If we found a match here, I think we should print a warning for the user that their database CA will be rotated alongside host CA (and vice versa). Otherwise they might not expect it and it will break their database access.

Reply:
Added